docs/markdown/Python/python-integrations/protobuf-python.md

```diff
@@ -72,6 +72,32 @@ protobuf>=3.12.1
 
 Pants will then automatically add these dependencies to your `protobuf_source` targets created in the next step.
 
+### Use alternative grpc plugins
+
+By default, the `grpcio` plugin is used to generate the services stubs. Pants can also generate services stubs using [grpclib](https://github.com/vmagamedov/grpclib).
+
+To use [grpclib](https://github.com/vmagamedov/grpclib), set `grpclib_plugin = true` in the `[python-protobuf]` scope. You probably also want to disable the `grpcio` plugin if you don't use it.
+
+```toml pants.toml
+[python-protobuf]
+grpclib_plugin = true
+grpcio_plugin = false
+```
+
+Don't forget to add the `grpclib` dependency to your requirements.
+
+It is recommended to share the same dependencies versions between the protobuf plugin and your code runtime dependency (see [Python Lockfile](doc:python-lockfiles#sharing-lockfiles-between-tools-and-code). The requirements for the `grpclib` plugin can be defined this way:
```

Comment:
This is relevant not just for grpclib but for grpcio too, so this (and the example of how to share dependency versions) could go before line 68 above (while keeping the example of explicitly listing the dependency versions too, as people may prefer that). I know that's not related to your change, but since you're here anyway...

I can also take a swipe at docs fixups after we merge this.

Reply:
I might be missing something, but isn't `grpcio` plugin downloaded as a static PEX package instead of a `PythonToolRequirementsBase` or similar? I don't see any option to set the resolve for it.

Reply:
Ah yes, build-time grpcio plugin is, and that happens automatically when you turn the `grpc` switch on for a target. But the runtime wheel *also* needs to be in your requirements, which is what this line is talking about: https://github.com/pantsbuild/pants/blob/0b28a298f7d003cd4ea3adf93b5ff59e844507e3/docs/markdown/Python/python-integrations/protobuf-python.md?plain=1#L69

So you are correct that the "sharing lockfiles" part is not relevant here. It is still relevant to say that the version here must be synced with the build time tool version, but it is annoying to explain how to do that (you have to look at/configure the version in the `grpc-python-plugin` subsystem and match that).

Possibly in a future change we should move away from downloading the binary `grpc_python_plugin` we build and host, and instead consume it from inside the grpcio wheel (which I believe provides it, since you can run `python -m grpc_tools.protoc` to invoke protoc+grpc). I think that should be possible. But that is for another day...

So anyway, for now what you have is fine as-is.